### PR TITLE
Capi score

### PR DESCRIPTION
**Pull request by mnunberg**

### CODE REVIEW

**Comment by mnunberg** [comment]
This PR contains a bunch of miscellaneous fixes:

1. Allow the C API to return scores
2. Fix some C API naming conventions
3. Make the C API return error messages and status information for indexing
4. Consolidate some long-arg forms into option structs
5. Fix some memory leaks
6. Provide a module level destructor so that we can use valgrind easier

File: src/forward_index.c [hunk 1]
```diff
@@ -238,15 +246,15 @@ int forwardIndexTokenFunc(void *ctx, const Token *tokInfo) {
...
+                                 tokCtx->fieldScore, tokCtx->fieldId, TOKOPT_F_COPYSTR);
       }
     }
   }
 
   if (tokInfo->phoneticsPrimary) {
     ForwardIndex_HandleToken(tokCtx->idx, tokInfo->phoneticsPrimary,
                              strlen(tokInfo->phoneticsPrimary), tokInfo->pos, tokCtx->fieldScore,
-                             tokCtx->fieldId, 0, 0, true);
+                             tokCtx->fieldId, TOKOPT_F_COPYSTR);
```
Comment (MeirShpilraien) [inline src/forward_index.c]: You are passing the same options here and above but it was not the same before the commit ...

File: src/redisearch_api.c [hunk 2]
```diff
@@ -123,17 +131,21 @@ static Document* RS_CreateDocument(const void* docKey, size_t len, double score,
...
   t_docId id = DocTable_GetIdR(&sp->docs, docId);
   if (id == 0) {
-    RedisModule_FreeString(NULL, docId);
-    return 0;
-  }
-  int rc = DocTable_DeleteR(&sp->docs, docId);
-  if (rc) {
-    sp->stats.numDocuments--;
-    return 1;
+    rc = 0;
```
Comment (MeirShpilraien) [inline src/redisearch_api.c]: rc is already zero, just do if(id != 0) without any else.

File: src/forward_index.c (hunk 1 above)
Comment (mnunberg) [reply src/forward_index.c]: The change is intentional. Without the change there is a memory leak (it makes more sense if you see the individual commit). Basically the commit frees the string at the source, and also makes sure that the string here is copied.

File: src/redisearch_api.c (hunk 2 above)
Comment (mnunberg) [reply src/redisearch_api.c]: I think I maybe intended for rc to be REDISMODULE_OK and REDISMODULE_ERR, like all the other APIs. Let me change this.

**Comment by mnunberg** [comment]
@MeirShpilraien I've addressed the comments. Should be good to go now

**Pull request merged**